services/fastfile_peeker/fastfile_peeker.rb

```diff
@@ -0,0 +1,55 @@
+require "fastfile_parser"
+require "digest"
+
+module FastlaneCI
+  # Utility class designed for parsing Fastfiles in a repo.
+  class FastfilePeeker
+    class << self
+      def cache
+        @cache ||= {}
+        return @cache
+      end
+
+      # Class method that finds the directory for the first Fastfile found given a root_path
+      # @param [String] root_path
+      # @return [String, nil] the path of the Fastfile or nil if no Fastfile found.
+      def fastfile_path(root_path: nil)
+        fastfiles = Dir[File.join(root_path, "fastlane/Fastfile")]
+        fastfiles = Dir[File.join(root_path, "**/fastlane/Fastfile")] if fastfiles.count == 0
+        fastfile_path = fastfiles.first
+        return fastfile_path
+      end
+
+      protected :cache, :fastfile_path
+
+      # @param [GitRepo] git_repo
+      # @param [String, nil] branch
+      # @param [String, nil] sha
+      # @return [Fastlane::FastfileParser]
+      def peek(git_repo: nil, branch: nil, sha: nil, cache: true)
+        cache_key = git_repo.git_config.git_url + (branch || sha)
```

Comment:
I think we should just remove the cache for now. It adds complexity and I don't think the performance gains are big enough to worry about right now.

Reply:
`peek` is being done each time the user selects a new `branch` in the `New Project` workflow. I can't imagine the user waiting to the whole repo being cloned each time he changes the branch from the selector in the interface. As I think that cache might be complex in certain scenarios, I think this must be keeped for this one

Reply:
I correct myself, as it has a direct dependency on `GitRepo` and it will be cloned whether the cache is used or not, I'll delete it until the `GitRepo` API allows to clone the repo out of the initializer.